apps/server/src/modules/transfer/export/resource-exporter.ts

```diff
@@ -0,0 +1,43 @@
+import { injectable } from 'inversify';
+
+import { RESOURCE_TYPE as FLOW_RESOURCE_TYPE } from '@flogo-web/plugins/flow-server';
+import {
+  ExportActionAgent,
+  Resource,
+  ResourceExportContext,
+} from '@flogo-web/lib-server/core';
+
+import { ResourceService } from '../../resources';
+import { AllContribsService } from '../../all-contribs';
+import { ResourcePluginRegistry } from '../../../extension';
+
+@injectable()
+export class ResourceExporter {
+  constructor(
+    private resourceService: ResourceService,
+    private pluginRegistry: ResourcePluginRegistry,
+    private allContribsService: AllContribsService
+  ) {}
+
+  async export(resourceId) {
+    const resource = await this.resourceService.findOne(resourceId);
+    const contributions = await this.allContribsService.allByRef();
+    const resourceExporter = this.pluginRegistry.resourceTypes.exporter(
```

Comment:
@fcastill if resourceTypes are always going to follow a structure, can we have a exporter extractor function at pluginRegistry level instead?

```suggestion
    const resourceExporter = this.pluginRegistry.getExporterForType()
```